src/participants/create_project.py

```diff
@@ -104,7 +104,11 @@ def get_repolinks(self, wid, project):
                 repolinks[repo].append(arch)
             if not repolinks[repo]:
                 del repolinks[repo]
-        return repolinks
+        if repolinks:
+            return repolinks
+        else:
+            # Creating a project with no repos makes no sense
+            raise RuntimeError("No suitable repos found in %s (must contain an arch in the name)" % project)
```

Comment:
We could use some tests to document(/test) the different types of behaviour we expect.

create_project gets called from VCSCOMMIT_BUILD, and its main task is to ensure a project exists for trigger_service to stash a package.  Whether it builds or not is a different matter and should rather be caught by do_wait_for_build.